doc/guides/modules.rst

```diff
@@ -10,14 +10,352 @@ system those are called *modules*. These modules must be integrated with the
+     an external project have its own practice of conveying how licensing
+     applies in the module code base, this practice may be accepted by and
+     be referred to in the Zephyr module, as long as licensing requirement,
+     for example OSI compliance, are satisfied.
+
+License policies
+================
+
+When creating a module repository a developer shall:
+
+* import the main license files, if they exist in the external project, and
+* document (for example in the module README or .yml file) the default license
+  that covers the module code base.
+
+License checks
+--------------
+
+License checks (via CI tools) shall be enabled on every pull request that
+adds new content in module repositories.
+
+
+Documentation requirements
+**************************
+
+All Zephyr module repositories shall include an .rst file documenting:
+
+* the scope and the purpose of the module
+* how the module integrates with Zephyr
+* the owner of the module repository
+* synchronization information with the external project (commit, SHA, version etc.)
+* licensing information as described in :ref:`modules_licensing`.
+
+The file shall be required for the inclusion of the module and the contained
+information should be kept up to date.
+
+
+Testing requirements
+********************
+
+All Zephyr modules should provide some level of **integration** testing,
+ensuring that the integration with Zephyr works correctly.
+Integration tests:
+
+* may be in the form of a minimal set of samples and tests that reside
+  in the Zephyr main tree
+* should verify a basic usage of the module (configuration,
+  functional APIs, etc.) that is integrated with Zephyr.
+* shall be built and executed (for example in QEMU) as part of
+  sanitycheck runs in pull requests that introduce changes in module
+  repositories.
+
+  .. note::
+
+     Vendor HALs are implicitly tested via Zephyr tests built or executed
+     on target platforms, so they do not need to provide integration tests.
+
+The purpose of integration testing is not to provide functional verification
+of the module; this should be part of the testing framework of the external
+project.
+
+Certain external projects provide test suites that reside in the upstream
+testing infrastructure but are written explicitly for Zephyr. These tests
+may (but are not required to) be part of the Zephyr test framework.
+
+Deprecating and removing modules
+*********************************
+
+Modules may be deprecated for reasons including, but not limited to:
+
+* Lack of maintainer-ship in the module
+* Licensing changes in the external project
+* Code base becoming obsolete
+* Vendor HAL restructuring that impacts the module code base
+
+The module information shall indicate whether a module is
+deprecated and the build system shall issue a warning
+when trying to build Zephyr using a deprecated module.
+
+Deprecated modules may be removed from the Zephyr default manifest
+after 2 Zephyr releases.
```

Comment:
What is the criteria for a module to be included in the Zephyr default manifest.    I'm guessing some level of testing is required and test cases be made available, but not seeing it in the above text.

Reply:
@kestewart at the beginning of this section we list the criteria for an external project to be a candidate for becoming a zephyr module. I guess this implies being included in the manifest.